pkg/content/package.go

```diff
@@ -0,0 +1,119 @@
+package content
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+type defaultPackageInspector struct {
+	packageInfo         chrootContentFunc
+	buildPackageCommand chrootCommandFunc
+	cmd                 *exec.Cmd
+
+	Packages []packages
+}
+
+type packages struct {
+	Package      string
+	Version      string
+	Architecture string
+	Description  string
+}
+
+// NewDefaultPackageInspector returns a new Version Inspector
+func NewDefaultPackageInspector() Inspector {
+	inspector := &defaultPackageInspector{}
+
+	return inspector
+}
+
+//buildPathChroot Builds the new PATH environment variable for executing
+//commands
+func buildPathChroot(mountPath string) string {
+
+	pathStr := os.Getenv("PATH")
+	pathSlice := strings.Split(pathStr, ":")
+	var newPath bytes.Buffer
+
+	newPath.WriteString("PATH=")
+	for i, binPath := range pathSlice {
+		if i != 0 {
+			newPath.WriteString(":")
+		}
+		newPath.WriteString(path.Join(mountPath, binPath))
+	}
+	return newPath.String()
+}
+
+//buildDpkgCommandChroot builds the command for DPKG based images
+func (s *defaultPackageInspector) buildDpkgCommandChroot(mountPath string) {
+	cmd := exec.Command(path.Join(mountPath, "/usr/bin/dpkg-query"), "-W",
+		"-f", "${binary:Package},${Version},${Architecture},${Description}---",
+		"--admindir", path.Join(mountPath, "/var/lib/dpkg"))
+	cmd.Env = []string{buildPathChroot(mountPath)}
+	s.cmd = cmd
+
+}
+
+//buildRpmCommandChroot builds the command for RPM based images
+func (s *defaultPackageInspector) buildRpmCommandChroot(mountPath string) {
+	cmd := exec.Command("rpm", "-qa", "--dbpath",
+		path.Join(mountPath, "/var/lib/rpm"), "--qf",
+		"%{NAME},%{VERSION},%{ARCH},%{DESCRIPTION}---")
+	cmd.Env = []string{buildPathChroot(mountPath)}
+	s.cmd = cmd
+}
+
+//getPackageInformation executes the appropriate command and parses the output
+func (s *defaultPackageInspector) getPackageInformation() error {
+
+	out, err := s.cmd.CombinedOutput()
+
+	if err != nil {
+		return fmt.Errorf("Unable to read package content")
+	}
+	pkgdata := strings.Replace(string(out), "\n", "", -1)
+	pkglist := strings.Split(pkgdata, "---")
+
+	for _, pkg := range pkglist {
+		if len(pkg) > 0 {
+			parts := strings.Split(pkg, ",")
+			pkgDescription := strings.Join(parts[3:], "")
+
+			s.Packages = append(s.Packages, packages{Package: parts[0],
+				Version: parts[1], Architecture: parts[2], Description: pkgDescription})
+		}
+	}
+
+	return nil
+}
+
+//Inspect is the primary execution entrypoint
+func (s *defaultPackageInspector) Inspect(mountPath string) error {
+	_, err := os.Stat(path.Join(mountPath, "/var/lib/rpm"))
```

Comment:
should we rely on the hard coded binary or resolve this via the path variable for the image?